source/dem/pw_linear_force.cc

```diff
@@ -95,48 +143,39 @@ PWLinearForce<dim>::calculate_pw_contact_force(
 template <int dim>
 std::tuple<Tensor<1, dim>, Tensor<1, dim>, Tensor<1, dim>, Tensor<1, dim>>
 PWLinearForce<dim>::calculate_linear_contact_force_and_torque(
-  const Parameters::Lagrangian::PhysicalProperties &physical_properties,
-  pw_contact_info_struct<dim> &                     contact_info,
-  const ArrayView<const double> &                   particle_properties)
+  pw_contact_info_struct<dim> &  contact_info,
+  const ArrayView<const double> &particle_properties)
 {
-  // Calculation of effective Young's modulus of the
-  // contact
-  double effective_youngs_modulus =
-    (physical_properties.youngs_modulus_particle *
-     physical_properties.youngs_modulus_wall) /
-    (physical_properties.youngs_modulus_wall *
-       (1 - physical_properties.poisson_ratio_particle *
-              physical_properties.poisson_ratio_particle) +
-     physical_properties.youngs_modulus_particle *
-       (1 - physical_properties.poisson_ratio_wall *
-              physical_properties.poisson_ratio_wall));
+  const unsigned int particle_type =
+    particle_properties[DEM::PropertiesIndex::type];
 
   // Calculation of normal and tangential spring and dashpot constants
   // using particle properties
   double normal_spring_constant =
     1.0667 * sqrt((particle_properties[DEM::PropertiesIndex::dp] / 2)) *
-    effective_youngs_modulus *
+    this->effective_youngs_modulus[particle_type] *
     pow((1.0667 * particle_properties[DEM::PropertiesIndex::mass] *
          contact_info.normal_relative_velocity *
          contact_info.normal_relative_velocity /
          (sqrt((particle_properties[DEM::PropertiesIndex::dp] / 2)) *
-          effective_youngs_modulus)),
+          this->effective_youngs_modulus[particle_type])),
         0.2);
   double tangential_spring_constant =
     1.0667 * sqrt((particle_properties[DEM::PropertiesIndex::dp] / 2)) *
-      effective_youngs_modulus *
+      this->effective_youngs_modulus[particle_type] *
       pow((1.0667 * particle_properties[DEM::PropertiesIndex::mass] *
            contact_info.tangential_relative_velocity *
            contact_info.tangential_relative_velocity /
            (sqrt((particle_properties[DEM::PropertiesIndex::dp] / 2)) *
-            effective_youngs_modulus)),
+            this->effective_youngs_modulus[particle_type])),
           0.2) +
     DBL_MIN;
   double normal_damping_constant = sqrt(
     (4 * particle_properties[DEM::PropertiesIndex::mass] *
      normal_spring_constant) /
-    (1 + pow((M_PI / (log(physical_properties.restitution_coefficient_wall) +
-                      DBL_MIN)),
+    (1 + pow((M_PI /
+              (log(this->effective_coefficient_of_restitution[particle_type]) +
```

Comment:
Try to gather these effective coefficient one and store them in a local variable instead of regathering them .